### ISSUE TITLE
aria-allowed-role throws violation when applying role grid to ul tag

### ISSUE THREAD

**Issue by curtbellew** [dequelabs/axe-core#4329]
### Product

axe-core

### Product Version

4.7.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
I put role="grid" on an unordered list - expected no violations returned from axe-core aria-allowed-role.
#### Actual
 The axe-core aria-allowed-role rule is reporting "ARIA role row is not allowed for given element".  
#### How to Reproduce
```
<ul role="grid">
<li role="row"><div role="gridcell">one</div></li>
<li role="row"><div role="gridcell">two</div></li>
<li role="row"><div role="gridcell">three</div></li>
<li role="row"><div role="gridcell">four</div></li>
</ul>
```
#### Additional context
I would expect this kind of violation returned if I had applied the role="grid" on something like a button but I can't justify it being thrown on a ul.

**Comment by straker** [comment]
Thanks for the issue. For valid role issues we follow the [HTML in ARIA spec](https://www.w3.org/TR/html-aria) which dictates which roles and attributes are allowed on what elements. Specifically for [ul and ol elements](https://www.w3.org/TR/html-aria/#el-ul) the `grid` role is not allowed.
